### PR TITLE
ENT-2966 | added flag `enable_compression`

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
https://openedx.atlassian.net/browse/ENT-2966

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/models.py [hunk 1]
```diff
@@ -2368,6 +2371,13 @@ class EnterpriseCustomerReportingConfiguration(TimeStampedModel):
         verbose_name=_("Delivery Method"),
         help_text=_("The method in which the data should be sent.")
     )
+    enable_compression = models.BooleanField(
+        default=True,
+        help_text=_(
+            "Specifies whether report should be compressed. Without compression files will not be password protected "
+            "or encrypted."
```
Comment (muhammad-ammar) [inline enterprise/models.py:2378]: I know compression is needed to set a password but how is encryption dependent on compression?

**Review by muhammad-ammar — APPROVED** [review]
👍 once build is green

File: enterprise/models.py [hunk 2]
```diff
@@ -2325,9 +2325,12 @@ class EnterpriseCustomerReportingConfiguration(TimeStampedModel):
         (DATA_TYPE_GRADE, DATA_TYPE_GRADE),
         (DATA_TYPE_COMPLETION, DATA_TYPE_COMPLETION),
         (DATA_TYPE_COURSE_STRUCTURE, DATA_TYPE_COURSE_STRUCTURE),
-
     )
 
+    # This list will specifies the Data Types that are allowed to send without compression.
+    # All other Data Types must be compressed.
```
Comment (sameenfatima78) [inline enterprise/models.py:2331]: Nit:
```python
# This list will specifies the Data Types that are allowed to send without compression.    
# All other Data Types must be compressed.
-> 
# Data types that are allowed to be sent without compression, all other data types must be compressed.

Comment (sameenfatima78) [reply enterprise/models.py:2331]: Also, shouldn't we handle the `invalid-name` warning?

File: enterprise/models.py (hunk 1 above)
Comment (sameenfatima78) [reply enterprise/models.py:2378]: shouldn't this change in model have a migration? I can't seem to find any migration file in the PR

File: enterprise/models.py [hunk 3]
```diff
@@ -2579,6 +2590,13 @@ def clean(self):
                     'Decrypted SFTP password must be set if the delivery method is SFTP.'
                 )
 
+        # Check compression is enabled for compression only Data Types.
+        if not self.enable_compression and self.data_type not in self.ALLOWED_NON_COMPRESSION_DATA_TYPES:
```
Comment (sameenfatima78) [inline enterprise/models.py:2594]: Maybe we can add parenthesis b/w conditions to add more readability

**Review by sameenfatima78 — CHANGES_REQUESTED** [review]
Left some comments, all good but migration file is missing

File: enterprise/models.py (hunk 2 above)
Comment (HammadAhmadWaqas) [reply enterprise/models.py:2331]: @sameenfatima78 actually its name is too long. the valid name should be less than 30 chars.

I am unable to think of any name that is less than 30 chars any, meaningful.
you can suggest any if you have any.

**Comment by HammadAhmadWaqas** [comment]
@sameenfatima78 could you please re-reveiw.

**Review by sameenfatima78 — APPROVED** [review]
LGTM!

**Pull request merged**